packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue

```diff
@@ -377,39 +363,76 @@ async function handleDecapodesPreview(data: any) {
 	if (response) {
 		const m = await getModel(response.id);
 		if (m) {
-			selectedModel.value = m;
-			previewHTML.value = `Decapode created ${m.id}`;
+			clonedState.value.modelId = m.id;
+			emit('append-output-port', {
+				label: 'Output',
+				state: cloneDeep(clonedState.value),
+				isSelected: false,
+				type: 'modelId',
+				value: [clonedState.value.modelId]
+			});
 		}
 		isProcessing.value = false;
 	}
 }
 
-/**
- * Editor initialization function
- * @param editorInstance	the Ace editor instance
- */
-async function initialize(editorInstance) {
-	editor.value = editorInstance;
+// The format of code blocks is a real pain to work with...
+async function getInputCodeBlocks() {
+	const codeAssetId: string = props.node.inputs?.[0]?.value?.[0];
+	if (!codeAssetId) return;
+	const codeAsset = await getCodeAsset(codeAssetId);
+	if (!codeAsset) return;
+
+	if (!codeAsset.id || !codeAsset.files) return;
+	let codeBlocks: CodeBlock[] = [];
+
+	const filteredFiles = Object.entries(codeAsset.files).filter(
+		(fileEntry) => fileEntry[1].dynamics?.block?.length > 0
+	);
+
+	const promises = filteredFiles.map(async (fileEntry) => {
+		const codeContent = (await getCodeFileAsText(codeAsset.id!, fileEntry[0])) ?? '';
+
+		// Assuming fileEntry[1].dynamics.block is an array
+		const inputCodeBlocksForFile = fileEntry[1].dynamics.block.map((block) => {
+			const { startRow, endRow } = extractDynamicRows(block);
+			const codeSnippet = extractCodeLines(codeContent, startRow, endRow);
+
+			return {
+				filename: fileEntry[0],
+				block,
+				codeLanguage: fileEntry[1].language,
+				codeContent: codeSnippet ?? '',
+				name: 'Code Block',
+				includeInProcess: true,
+				type: CodeBlockType.INPUT
+			};
+		});
+
+		return inputCodeBlocksForFile;
+	});
+
```

Comment:
Codeblocks are a pain to work with I want to convert something like this:
```
{
  "files": {
    "chime-1.1.5/.dockerignore": {
      "dynamics": {
        "name": "test",
        "description": "",
        "block": [
          "L1-L9"
        ]
      }
    },
    "chime-1.1.5/.bldr.jl": {
      "dynamics": {
        "name": "test",
        "description": "",
        "block": [
          "L28-L33",
          "L1-L9"
        ]
      }
    }
  }
}
```

 into a more workable format, the format being the CodeBlock interface

Reply:
Feel free to update the back end enpoints to send data better formatted for your needs.
That is why it is there.

Reply:
will do, might do i in another PR though as I finalize everything